stake/src/validator_state.rs

```diff
@@ -16,8 +21,51 @@ pub enum ValidatorState {
     Slashed,
 }
 
+impl Protobuf<pb::ValidatorState> for ValidatorState {}
+
+impl From<ValidatorState> for pb::ValidatorState {
+    fn from(v: ValidatorState) -> Self {
+        pb::ValidatorState {
+            unbonding_epoch: match v {
+                ValidatorState::Unbonding { unbonding_epoch } => Some(unbonding_epoch),
+                _ => None,
+            },
+            state: match v {
+                ValidatorState::Inactive => pb::validator_state::ValidatorStateEnum::Inactive,
+                ValidatorState::Active => pb::validator_state::ValidatorStateEnum::Active,
+                ValidatorState::Unbonding { unbonding_epoch } => {
+                    pb::validator_state::ValidatorStateEnum::Unbonding
+                }
+                ValidatorState::Slashed => pb::validator_state::ValidatorStateEnum::Slashed,
+            } as i32,
+        }
+    }
+}
+
+impl TryFrom<pb::ValidatorState> for ValidatorState {
+    type Error = anyhow::Error;
+    fn try_from(v: pb::ValidatorState) -> Result<Self, Self::Error> {
+        Ok(
+            match pb::validator_state::ValidatorStateEnum::from_i32(v.state)
+                .ok_or_else(|| anyhow::anyhow!("missing validator state"))?
+            {
+                pb::validator_state::ValidatorStateEnum::Inactive => ValidatorState::Inactive,
+                pb::validator_state::ValidatorStateEnum::Active => ValidatorState::Active,
+                pb::validator_state::ValidatorStateEnum::Unbonding => ValidatorState::Unbonding {
+                    unbonding_epoch: v
+                        .unbonding_epoch
+                        .ok_or_else(|| anyhow::anyhow!("missing unbonding epoch"))?,
+                },
+                pb::validator_state::ValidatorStateEnum::Slashed => ValidatorState::Slashed,
+            },
+        )
+    }
+}
+
 /// The name of a validator state, as a "C-style enum" without the extra information such as the
 /// `unbonding_epoch`.
+/// TODO: is this necessary now? Is `ValidatorStatus` necessary or are we decomposing into different
+/// paths in the JMT?
```

Comment:
idk? it seems possibly useful to bundle all the state data into one path, so some client can authenticate a proof of inclusion for whatever the state is

Reply:
The `ValidatorStatus` struct also includes the voting power (and another copy of the identity key). It's not clear to me yet how we'll handle voting power in the staking component yet. It might make more sense to serialize the entire `ValidatorStatus` to a single path rather than splitting power and state, idk, I'll figure it out when I get there